flocker/acceptance/testtools.py

```diff
@@ -1091,6 +1108,100 @@ def clean(cluster):
     return decorator
 
 
+def is_container_agent_running(node):
+    """
+    Check if the container agent is running on the specified node.
+
+    :param Node node: the node to check.
+    :return Deferred[bool]: a Deferred that will fire when
+        with whether the container agent is runnning.
+    """
+    d = node.run_script("service_running", "flocker-container-agent")
+
+    def not_existing(failure):
+        failure.trap(ProcessTerminated)
+        return False
+    d.addCallbacks(lambda result: True, not_existing)
+    return d
+
+
+def set_container_agent_enabled_on_node(node, enabled):
+    """
+    Ensure the container agent is enabled/disabled as specified.
+
+    :param Node node: the node on which to ensure the container
+        agent's state
+    :param bool enabled: True to ensure the container agent
+        is enabled and running, false to ensure the opposite.
+    :return Deferred[None]: a Deferred that will fire when
+        the container agent is in the desired state.
+    """
+    if enabled:
+        d = node.run_script("enable_service", "flocker-container-agent")
+    else:
+        d = node.run_script("disable_service", "flocker-container-agent")
+    # If the agent was disabled We have to reboot to clear the control cache.
+    # If we want to avoid the reboot we could add an API to do this.
+    # XXX: does this need to reboot the control node? the node that
+    # changed?
```

Comment:
It should reboot node that container agent is on. We have concept of eras, rebooting gives you a new era so for practical purposes the rebooted node is a completely new node with no cached history (even though the old data is still in control service, it's not used.)